### ISSUE TITLE
Suppress unwanted analyzer warnings on Linux builds

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#11978]
Change https://github.com/dotnet/roslyn/pull/11956 fixed analyzer loading issues on Linux. Now we are getting warnings that should be suppressed.

**Comment by srivatsn** [comment]
These are warning that are only in linux builds and not in windows build? That sounds like maybe a configuration issue. Is there a reason to suppress these?

**Comment by mavasani** [comment]
What is the correct way to build Roslyn on mac? I tried running `./cibuild.sh --nocache --debug` but that gives me following error:

```
error: Unable to load the service index for source https://dotnet.myget.org/F/dotnet-coreclr/api/v3/index.json.
error:   The type initializer for 'Crypto' threw an exception.
error:   The type initializer for 'CryptoInitializer' threw an exception.
error:   Unable to load DLL 'System.Security.Cryptography.Native': The specified module could not be found.
error:    (Exception from HRESULT: 0x8007007E)
Restore CrossPlatform.sln
error: Unable to load the service index for source https://dotnet.myget.org/F/dotnet-coreclr/api/v3/index.json.
error:   The type initializer for 'Crypto' threw an exception.
error:   The type initializer for 'CryptoInitializer' threw an exception.
error:   Unable to load DLL 'System.Security.Cryptography.Native': The specified module could not be found.
error:    (Exception from HRESULT: 0x8007007E)
Restore failed
```

**Comment by tmat** [comment]
Do you have OpenSSL installed?

**Comment by mavasani** [comment]
Yes, that unblocked me and I can repro it locally. How do I build the csproj files individually?

**Comment by mavasani** [comment]
Unable to execute MSBuild: `-bash: Binaries/toolset/roslyn.mac.8/MSBuild.exe: cannot execute binary file`

**Comment by tmat** [comment]
See https://github.com/dotnet/roslyn/blob/master/Makefile#L39

**Comment by mavasani** [comment]
Thanks!

**Comment by mavasani** [comment]
So it seems like the ruleset include over [here](https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/CodeAnalysisRules.ruleset#L4) is not resolving to an existing file on mac/linux machines, and hence all rule suppressions in the included ruleset are ignored. If I change the include path to have separator char `/` then it builds fine without warnings.  It seems like [this](http://source.roslyn.io/#Microsoft.CodeAnalysis/RuleSet/RuleSetInclude.cs,03c5b86cce97cb7d,references) code needs to handle this and normalize the paths before checking if it exists.

**Comment by mavasani** [comment]
@srivatsn @tmat It seems that NormalizePath APIs in FileUtilities also retain the Window style directory separator chars in the include file: `Could not locate the rule set file '/mnt/j/workspace/dotnet_roslyn/master/linux_debug_prtest/src/Compilers/Core/..\..\..\build\Rulesets\Roslyn_BuildRules.ruleset'`

I am wondering if we need to fix the code that computes the include path to replace all Window style directory separator char `\\` to AltDirectorySeparatorChar `/` OR require that ruleset Include always use AltDirectorySeparatorChar to build on non-Windows OS? I am tending to think that it must be the latter.

**Comment by mavasani** [comment]
I took the former approach, PR here: https://github.com/dotnet/roslyn/pull/13350
